Remove the site survey banner

oikeme opened the pull request:
The site survey banner is no longer needing to be displayed and can be removed.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

* Removes the agency survey banner which is no longer needed

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Ensure that the survey JITM banner is no longer displayed

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

oikeme committed:
Remove the site survey banner

The site survey banner is no longer needing to be displayed and can be removed.

yashwin reviewed (commented):
How about we set `hideBanner: true` in the banner object instead of removing it? Since this is a survey banner, it could be needed again. WDYT?

oikeme committed:
Revert "Remove the site survey banner"

This reverts commit 04871bb2dbef9bd5916cf8cb780ea57edcdb8c5c.

oikeme committed:
Update index.tsx

Use hide banner in place of removing the banner entirely in case it needs to be reused in the future

oikeme committed:
Merge branch 'trunk' into update/remove-agency-survey-banner

yashwin reviewed (approved):
LGTM
